plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java

```diff
@@ -95,6 +100,10 @@
 
 public class TransactionLogAccess
 {
+    private static final Pattern CLASSIC_CHECKPOINT = Pattern.compile("(\\d*)\\.checkpoint\\.parquet");
+    private static final Pattern MULTI_PART_CHECKPOINT = Pattern.compile("(\\d*)\\.checkpoint\\.(\\d*)\\.(\\d*)\\.parquet");
+    private static final Pattern V2_CHECKPOINT = Pattern.compile("(\\d*)\\.checkpoint\\.[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}\\.(json|parquet)");
```

Comment:
The number of digits isn't defined in Delta protocol as far as I know.

Reply:
Yes, does not look like it's part of the protocol. It's defined in the code, so there is likely an implicit convention here https://github.com/delta-io/delta/blob/01c0ef91565d9a3e06fb9f62295f7c80fba09351/kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/FileNames.java#L96-L107
It looks like the first group is always 20 digits (which might be safe to assume, since the checkpoints are snapshots for a commit version, and commit files are always 20 digits), while for multi-part checkpoints the two final groups are 10 digits each.

Reply:
Created https://github.com/delta-io/delta/issues/2842 to get a clarification.